xbmc/utils/SeekHandler.cpp

```diff
@@ -19,51 +19,132 @@
  */
 
 #include "SeekHandler.h"
+#include "guilib/LocalizeStrings.h"
 #include "GUIInfoManager.h"
 #include "Application.h"
+#include "settings/AdvancedSettings.h"
+#include "settings/Settings.h"
+#include "settings/lib/Setting.h"
+#include "utils/log.h"
+#include "utils/StringUtils.h"
 
 CSeekHandler::CSeekHandler()
-: m_requireSeek(false),
-  m_percent(0.0f)
+: m_seekDelay(500),
+  m_requireSeek(false),
+  m_percent(0.0f),
+  m_percentPlayTime(0.0f),
+  m_analogSeek(false),
+  m_seekStep(0)
 {
 }
 
+CSeekHandler::~CSeekHandler()
+{
+}
+
+CSeekHandler& CSeekHandler::Get()
+{
+  static CSeekHandler instance;
+  return instance;
+}
+
 void CSeekHandler::Reset()
 {
   m_requireSeek = false;
   m_percent = 0;
+  m_seekStep = 0;
+  m_seekDelay = CSettings::Get().GetInt("videoplayer.seekdelay");
+
+  std::vector<CVariant> seekSteps = CSettings::Get().GetList("videoplayer.seeksteps");
+  m_forwardSeekSteps.clear();
+  m_backwardSeekSteps.clear();
+  for (std::vector<CVariant>::iterator it = seekSteps.begin(); it != seekSteps.end(); ++it) {
+    int stepSeconds = (*it).asInteger();
+    if (stepSeconds < 0)
+      m_backwardSeekSteps.insert(m_backwardSeekSteps.begin(), stepSeconds);
+    else
+      m_forwardSeekSteps.push_back(stepSeconds);
+  }
 }
 
-void CSeekHandler::Seek(bool forward, float amount, float duration)
+int CSeekHandler::GetSeekSeconds(bool forward)
 {
+  m_seekStep = m_seekStep + (forward ? 1 : -1);
+
+  int seconds = 0;
+  if (m_seekStep > 0)
+  {
+    if ((size_t)m_seekStep <= m_forwardSeekSteps.size())
+      seconds = m_forwardSeekSteps.at(m_seekStep - 1);
+    else
+      seconds = m_forwardSeekSteps.back() * (m_seekStep - m_forwardSeekSteps.size() + 1);
```

Comment:
a comment might be nice to have here so that others see what this is all about more easily (when exceeding the selected amount of steps repeat/sum up the last step size)